akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -36,6 +36,7 @@ object ActorMaterializer {
    * `namePrefix-flowNumber-flowStepNumber-stepName`.
    */
   def apply(materializerSettings: Option[ActorMaterializerSettings] = None, namePrefix: Option[String] = None)(implicit context: ActorRefFactory): ActorMaterializer = {
+    require(namePrefix.forall(_.trim.nonEmpty), s"namePrefix has to be a non-empty string, got $namePrefix")
```

Comment:
I wonder if we should be slightly more careful to not allocate unnecessary objects here:
```
namePrefix match {
  case Some(prefix) => if (prefix.trim.isEmpty) throw new IllegalArgumentException(s"namePrefix has to be a non-empty string, got $namePrefix")
  case None => // ok
}
```